src/utils/fs.js

```diff
@@ -95,4 +95,47 @@ export function waitUntil(predicateFn, callbackFn, timeout, timeoutFn) {
   };
 
   resultFn();
+}
+
+// array of all the fsReadyFunctions we need to call and proxy function to call them
+const _fsReadyFunctions = [];
+function proxiedFsReady() {
+  for( let x=0; x<_fsReadyFunctions.length; x++ ){
+    try {
+      _fsReadyFunctions[x]();
+    }catch( error ){
+      console.warn( "Proxied _fs_ready function threw error", error );
+    }
+  }
+}
+
+/**
+ * Add an _fs_ready function on window, but in a way that honors existing ones that are setup
+ * and ones that might be setup afterwards
+ * @param callbackFn Function to call when _fs_ready would be called
+ *
+ */
+export function registerFsReady( callbackFn ) {
+  // first if we already have fullstory, then call it back
+  if( hasFs() ){
```

Comment:
Perhaps we need an `isFsReady` utility function that encapsulates the above checks.

Reply:
So it is hasFs && getCurrentSessionURL logic for isFsReady?

Reply:
Right like `hasFs() && typeof window[window._fs_namespace].getCurrentSessionURL === 'function'`.  This means for us to accurately test code, we'll have to add some form of `setTimeout()` that assigns the function to `FS` and possibly returns a mock string value.  This would be part of the [FS jest mock ](https://github.com/fullstorydev/integration-examples/blob/master/__mocks__/fs.js).

Reply:
I just added a new isFsReady function for you to take a look at and changed this to call it and added a test function for it.